packages/core/js-client/build.ts

```diff
@@ -0,0 +1,104 @@
+import path, { dirname } from 'path';
+import type { InlineConfig, PluginOption } from 'vite';
+import { build } from 'vite';
+import { builtinModules, createRequire } from 'module';
+import tsconfigPaths from 'vite-tsconfig-paths';
+import inject from '@rollup/plugin-inject';
+import stdLibBrowser from 'node-stdlib-browser';
+import { fileURLToPath } from 'url';
+import { rm, rename } from 'fs/promises';
+import { replaceCodePlugin } from 'vite-plugin-replace';
+import pkg from './package.json' assert { type: 'json' };
+import libAssetsPlugin from '@laynezh/vite-plugin-lib-assets';
+
+const require = createRequire(import.meta.url);
+
+const commonConfig = (isNode: boolean): InlineConfig & Required<Pick<InlineConfig, 'build'>> => {
+    const esbuildShim = require.resolve('node-stdlib-browser/helpers/esbuild/shim');
+    return {
+        build: {
+            target: 'modules',
+            minify: 'esbuild',
+            lib: {
+                entry: './src/index.ts',
+                name: 'js-client',
+                fileName: `${isNode ? 'node' : 'browser'}/index`,
+            },
+            outDir: './dist',
+            emptyOutDir: false,
+            ...(isNode ? {
+                rollupOptions: {
+                    external: [...builtinModules, ...builtinModules.map(bm => `node:${bm}`)],
+                    plugins: [
+                        // @ts-ignore
+                        inject({
+                            self: 'global',
+                            'WorkerScope': ['worker_threads', '*'],
+                            'Worker': ['worker_threads', 'Worker'],
+                            'isMainThread': ['worker_threads', 'isMainThread'],
+                        })
+                    ]
+                }
+            } : {
+                rollupOptions: {
+                    plugins: [
+                        {
+                            // @ts-ignore
+                            ...inject({
+                                global: [esbuildShim, 'global'],
+                                process: [esbuildShim, 'process'],
+                                Buffer: [esbuildShim, 'Buffer']
+                            }), enforce: 'post'
+                        }
+                    ],
+                }
+            })
+        },
+        plugins: [tsconfigPaths(), libAssetsPlugin({
+            include: ['**/*.wasm*', '**/marine-worker.umd.cjs*'],
+            publicUrl: '/',
+        }), ...(isNode ? [replaceCodePlugin({
+        replacements: [
+            { from: 'require(`./${file}.js`)', to: 'require(`./linux.js`)' },
+            {
+                from: 'const { name, version } = req(\'../../package.json\')',
+                to: 'const { name, version } = { name: \'ssdp\', version: \'4.0.4\' }'
+            },
+            { from: 'eval("require")("worker_threads")', to: 'WorkerScope' },
+            { from: 'eval("require")("worker_threads")', to: 'WorkerScope' },
+        ]
```

Comment:
If I am not mistaken you agreed to add at least some comments in this file at least here so it will be less WTFs in the future about why it's needed and when it will not be needed anymore potentially